### PR TITLE
Add support for serialized/transactional read in 'Client.get{,_multi}'.

### LINKED ISSUE #1859: Datastore transactions do not work without context manager
[The docs][docs] state that you can use transactions without a context manager, but this is not exactly true because you need to resort to some private methods on `Client`:

* [this file][context] uses the context manager and produces correct output
* [this file][broken] uses transactions as described by the aforementioned docs and fails the assertion
* [this file][working] uses transactions along with calls to `client._push_batch` and `client._pop_batch` and produces correct output

The issue is evident in [`Client.get_multi`][get_multi] and [`Client.put_multi`][put_multi] where `transaction` and `current` are `None` unless a `Batch` or `Transaction` has been pushed to the `_batch_stack` and the only time that happens is when you [`__enter__`][enter] a `Batch`.

[docs]: https://googlecloudplatform.github.io/gcloud-python/stable/datastore-transactions.html#gcloud.datastore.transaction.Transaction
[context]: https://gist.github.com/Bogdanp/a86192d1a2a37dd48df5ebdbb39a49d1#file-context-py-L20
[broken]: https://gist.github.com/Bogdanp/a86192d1a2a37dd48df5ebdbb39a49d1#file-failing-py-L20
[working]: https://gist.github.com/Bogdanp/a86192d1a2a37dd48df5ebdbb39a49d1#file-passing-py-L20
[get_multi]: https://github.com/GoogleCloudPlatform/gcloud-python/blob/master/gcloud/datastore/client.py#L284
[put_multi]: https://github.com/GoogleCloudPlatform/gcloud-python/blob/master/gcloud/datastore/client.py#L336
[enter]: https://github.com/GoogleCloudPlatform/gcloud-python/blob/master/gcloud/datastore/batch.py#L273

### PR DESCRIPTION
**Pull request by tseaver**
Closes #1859.

### CODE REVIEW

**Comment by tseaver** [comment]
@daspecster PTAL

File: system_tests/datastore.py [hunk 1]
```diff
@@ -432,6 +432,43 @@ def test_transaction(self):
         self.case_entities_to_delete.append(retrieved_entity)
         self.assertEqual(retrieved_entity, entity)
 
+    def test_transaction_via_explicit_begin_get_commit(self):
```
Comment (daspecster) [inline system_tests/datastore.py]: Should we make this test more generic and less tailored to the specific use case that was reported in the issue? I'm good either way just something I was thinking about.

**Comment by daspecster** [comment]
@tseaver LGTM as long as you're ok with my comment.

File: system_tests/datastore.py (hunk 1 above)
Comment (tseaver) [inline system_tests/datastore.py]: I stripped out the threading complexity from the issue:  I guess I could simplify the testcase method a little further.

Comment (daspecster) [inline system_tests/datastore.py]: I'm ok either way, I was just thinking the whole bank transaction premise.

**Pull request merged**